### PR TITLE
PLENTY-431 Export variant id of cheapest variant

### PR DESCRIPTION
**Pull request by TobiasGraml11**

### CODE REVIEW

File: tests/MockData/Pim/Variations/response_with_no_main_variation.json [hunk 1]
```diff
@@ -2829,11 +2829,11 @@
       "markets": null,
       "salesPrices": [
         {
-          "salesPriceId": 1,
+          "salesPriceId": 0,
           "price": 79
         },
         {
-          "salesPriceId": 2,
+          "salesPriceId": 1,
```
Comment (dblaichinger) [inline tests/MockData/Pim/Variations/response_with_no_main_variation.json]: Why was this necessary? Also using a test response without main variation seems not the right case here, how about using `response_for_lowest_price_test.json` instead?

File: src/Wrapper/Product.php [hunk 2]
```diff
@@ -318,9 +318,9 @@ protected function processVariations(bool $checkAvailability = true): int
             $this->item->addProperty($packageSizeProperty);
         }
 
-        if ($variationId) {
+        if ($this->cheapestVariationId || $variationId) {
             $variationIdProperty = new Property('variation_id');
-            $variationIdProperty->addValue((string)$variationId);
+            $variationIdProperty->addValue((string)($this->cheapestVariationId ?? $variationId));
```
Comment (dblaichinger) [inline src/Wrapper/Product.php]: Extracting the value in an own line would help improve readability as the amount of brackets is quite high here.

**Review by dblaichinger — CHANGES_REQUESTED** [review]

File: tests/MockData/Pim/Variations/response_with_no_main_variation.json (hunk 1 above)
Comment (TobiasGraml11) [reply tests/MockData/Pim/Variations/response_with_no_main_variation.json:2836]: This was needed, to set the second variant as cheapest. Otherwise the test would be irrelevant.

I used this data, as it was already used by the previous test, but I switched it now.

Comment (TobiasGraml11) [reply tests/MockData/Pim/Variations/response_with_no_main_variation.json]: This was a leftover from a different test adaption without removing the main variant id implementation

**Review by dblaichinger — APPROVED** [review]
